File: python/largest_number_at_least_twice_of_others.py
from typing import List, Tuple


class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        largest, largestIndex = self.getMax(nums)
        for number in nums:
            if number != largest and 2 * number > largest:
                return -1
        return largestIndex

    def getMax(self, array: List[int]) -> Tuple[int, int]:
        largest, largestIndex = -float('inf'), -1
        for index, element in enumerate(array):
            if element > largest:
                largest = element
                largestIndex = index
        return largest, largestIndex

File: src/LargestNumberAtLeastTwiceOfOthers.java
public class LargestNumberAtLeastTwiceOfOthers {
    public int dominantIndex(int[] nums) {
        int largestIndex = largestIndex(nums), largest = nums[largestIndex];
        for (int number : nums) {
            if (number != largest && 2 * number > largest) {
                return -1;
            }
        }
        return largestIndex;
    }

    private int largestIndex(int[] array) {
        int largest = Integer.MIN_VALUE, largestIndex = -1;
        for (int index = 0 ; index < array.length ; index++) {
            if (array[index] > largest) {
                largest = array[index];
                largestIndex = index;
            }
        }
        return largestIndex;
    }
}
